framework/src/media/audio/audio_manager.c

```diff
@@ -952,18 +1035,52 @@ int start_audio_stream_out(void *data, unsigned int frames)
 		medvdbg("Resume the output audio card!!\n");
 	}
 
-	if (card->resample.necessary) {
-		resampled_frames = resample_stream_out(card, data, frames);
+	if ((device_channel_num = pcm_get_channels(card->pcm)) == 0) {
+		meddbg("Fail to get channel number\n");
+		goto error_with_lock;
+	}
+	rechanneling_ratio = (float)device_channel_num / (float)card->resample.user_channel;
+
+	if (card->resample.ratio != 1) {
```

Comment:
I think you'd better add 2 functions, expand_channels(src, dst), collapse_channels(src, dst)
and below '2' can be calculated based on these values.
Then we can support re-channeling fully(next PR)

Reply:
Please add TODO here

Reply:
TODO was added.